import java.util.Scanner;

public class String_Reverse_Words_in_a_String {
    static void main(String[] args) {
            Scanner sc = new Scanner(System.in);

            String str = sc.nextLine();
            String[] words = str.split(" ");

            for (int i = words.length - 1; i >= 0; i--) {
                System.out.print(words[i] + " ");
            }
        }
    }